src/plots/plots.js

```diff
@@ -101,10 +104,14 @@ plots.resize = function(gd) {
 
             Registry.call('relayout', gd, {autosize: true}).then(function() {
                 gd.changed = oldchanged;
-                resolve(gd);
+                // Only resolve if a new call hasn't been made!
+                if(gd._resolveResize === resolve) resolve(gd);
```

Comment:
Nice! Took me a while and drawing some timing diagrams to be sure I agreed with this, but I think you've got it 🏆

Only thing I might suggest is when we finally get to this `resolve(gd)`, should we also `delete gd._resolveResize`? Otherwise the next time someone resizes 10 minutes later, there will still be a `resolveLastResize(p)` call. In principle it's OK to resolve the same promise more than once, but it feels nicer not to, also then we don't need to worry about it during `purge`.

Reply:
The old promise resolver is now deleted in https://github.com/plotly/plotly.js/pull/4392/commits/5ee431edc89b69c38cfabb008cf84b35b14f2a8f !